common/mb_odometry.c

```diff
@@ -33,12 +33,12 @@ void mb_initialize_odometry(mb_odometry_t* mb_odometry, float x, float y, float
 *
 *******************************************************************************/
 void mb_update_odometry(mb_odometry_t* mb_odometry, mb_state_t* mb_state){
-    float diff = mb_state->yaw_delta - mb_state->gyro_heading_delta;
+    float diff = mb_state->yaw_delta - mb_state->encoder_yaw_delta;
```

Comment:
remove the variable "gyro_yaw_delta", cause I found that the gyro_heading_delta is actually same as yaw_delta, meaning we were only using the heading from imu. To do the fusion of heading from imu and encoder, declare a new variable named "encoder_yaw_delta"